src/Autumn/compile.jl

```diff
@@ -4,11 +4,140 @@ module Compile
 using ..AExpressions
 using ..Program
 
+export compileToJulia, toRepr
+
+struct AutumnCompileError <: Exception
+  msg
+end
+
+fixedSymbols = [:+, :-, :/, :*, :&&, :||, :>=, :<=, :>, :<, :(==)]
+
+AutumnCompileError() = AutumnCompileError("")
+
 "Compile `aexpr` into `program::Program`"
-function compiletojulia(::AExpr)::AProgram
-  # Do type inference
-  # Get external values
-  # Alot
+function compileToJulia(aexpr::AExpr) #::AProgram
+
+  ### DATA ###
+  data = Dict([("historyVars" => []),
+               ("externalVars" => []),
+               ("initnextVars" => []),
+               ("initnextOther" => []),
+               ("types" => [])])
+
+  ### HELPER FUNCTION ###
+  function toRepr(expr::AExpr, parent=Nothing)::String
+    if expr.head == :if
```

Comment:
I think this could be expressed more concisely/readably using pattern matching.  See pattern matching in `sexpr.jl`